openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala

```diff
@@ -539,10 +553,16 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas {
         case s: LazySchema       => LazySchema(s.name, updatedDocs(s.value))
         case s: OneOf            => s.copy(example = Some(exampleJson))
       }
-    new JsonSchema(
-      schema.ujsonSchema,
-      updatedDocs(schema.docs)
-    )
+    schema match {
+      case s: RecordReference[A] =>
+        println(s"RECORDREFERENCE ${s.name}")
+        s.withName(s.name + "$").withExample(exampleJson)
```

Comment:
This is a clever idea, but I'm not sure it'll actually work unless the name mangling is made more complicated.

Consider the case where the same `JsonSchema` is re-used twice to create two other schemas that just each have their examples changed. Since both calls for adding the example will use `withExampleJsonSchema`, they'll both end up adding the same `$` to the initial schema's name. We're now back where we started with a a collision in the schema name.